<launch>

  <arg name="rviz" default="true" />
  <arg name="scanPeriod" default="0.1" />

  <node pkg="loam_velodyne" type="multiScanRegistration" name="multiScanRegistration" output="screen">
    <param name="lidar" value="HDL-64E"/> <!-- options: VLP-16  HDL-32  HDL-64E -->
    <param name="scanPeriod" value="$(arg scanPeriod)" />

    <remap from="/multi_scan_points" to="/kitti/velo/pointcloud"/>
  </node>

  <node pkg="loam_velodyne" type="laserOdometry" name="laserOdometry" output="screen" respawn="true">
    <param name="scanPeriod" value="$(arg scanPeriod)" />
  </node>

  <node pkg="loam_velodyne" type="laserMapping" name="laserMapping" output="screen">
    <param name="scanPeriod" value="$(arg scanPeriod)" />
  </node>

  <node pkg="loam_velodyne" type="transformMaintenance" name="transformMaintenance" output="screen">
  </node>

  <node pkg="loam_kitti_tf" type="static_tf_pub_2" name="static_tf_pub_2" output="screen">
  </node>

  <node pkg="loam_kitti_tf" type="dense_point_pub" name="dense_point_pub" output="screen">
   <!-- param name="/cutoff_depth" value="10"/ -->
    <remap from="/loam/camera_color_left/image_depth" to="/kitti/camera_color_left/depth"/>
    <remap from="/loam/camera_color_left/image_raw" to="/kitti/camera_color_left/image_raw"/>
    <remap from="/loam/camera_color_left/camera_info" to="/kitti/camera_color_left/camera_info"/>
    <remap from="/dense_point" to="/kitti/camera_color_left/depth_points"/>
  </node>


  <node pkg="loam_kitti_tf" type="color_point_cloud" name="color_point_cloud" output="screen">
   <remap from="/pointcloud" to="/velodyne_cloud_registered"/>
   <!-- remap from="/pointcloud" to="/velodyne_cloud_3"/ !-->
   <!-- remap from="/pointcloud" to="/kitti/velo/pointcloud"/ !-->
   <remap from="/image_raw" to="/kitti/camera_color_left/image_raw"/>
   <remap from="/camera_info" to="/kitti/camera_color_left/camera_info"/>
   <remap from="/color_points" to="/kitti/camera_color_left/depth_sparse_points"/>
  </node>


  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find loam_kitti_tf)/rviz_cfg/loam_kitti_cfg_v2.rviz" />
  </group>

</launch>
